File: autorization_oauth2.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Net;
using System.Security.Cryptography;
using System.Text.RegularExpressions;
using Newtonsoft.Json;

namespace Bearer
{

    public class Bearer
    {
        public string scope { get; set; }
        public string token_type { get; set; }
        public string expires_in { get; set; }
        public string refresh_token { get; set; }
        public string access_token { get; set; }
    }

    class Program
    {
        static void Main(string[] args)
        {
            string consumerKey = "260392633753-18hl37euj7tpdk4pp4c1happd80d2o9t.apps.googleusercontent.com";
            string consumerSecret = "4aQUzmnt-08DpxHP9Cqbi7gS";
            string accessToken;
            
            byte[] byte1 = Encoding.ASCII.GetBytes("grant_type=client_credentials");

            HttpWebRequest bearerReq = WebRequest.Create("AIzaSyDK48i45yKE-OfYiVrsZtdfBzXYP1YyiSI") as HttpWebRequest;
            bearerReq.Accept = "application/json";
            bearerReq.Method = "POST";
            bearerReq.ContentType = "application/x-www-form-urlencoded";
            bearerReq.ContentLength = byte1.Length;
            bearerReq.KeepAlive = false;
            bearerReq.Headers.Add("Authorization", "Basic " + Convert.ToBase64String(Encoding.Default.GetBytes(consumerKey + ":" + consumerSecret)));
            Stream newStream = bearerReq.GetRequestStream();
            newStream.Write(byte1, 0, byte1.Length);

            WebResponse bearerResp = bearerReq.GetResponse();
            
            using (var reader = new StreamReader(bearerResp.GetResponseStream(), Encoding.UTF8))
            {
                var response = reader.ReadToEnd();
                Bearer bearer = JsonConvert.DeserializeObject<Bearer>(response);
                accessToken = bearer.access_token;
            }

            Console.WriteLine(accessToken);
            Console.Read();
        }
    }
}

File: autorization_oauth2.py
 __author__ = 'bdm4'

import requests, json

token_url = "https://api.byu.edu/token"

test_api_url = "AIzaSyDK48i45yKE-OfYiVrsZtdfBzXYP1YyiSI"    #API KEY / API URL

#client (application) credentials on apim.byu.edu
client_id = '260392633753-18hl37euj7tpdk4pp4c1happd80d2o9t.apps.googleusercontent.com'  # ID client
client_secret = '4aQUzmnt-08DpxHP9Cqbi7gS'                                              # Client secret

#step A, B - single call with client credentials as the basic auth header - will return access_token
data = {'grant_type': 'client_credentials'}

access_token_response = requests.post(token_url, data=data, verify=False, allow_redirects=False, auth=(client_id, client_secret))

print access_token_response.headers
print access_token_response.text

tokens = json.loads(access_token_response.text)

print "access token: " + tokens['access_token']

#step B - with the returned access_token we can make as many calls as we want

api_call_headers = {'Authorization': 'Bearer ' + tokens['access_token']}
api_call_response = requests.get(test_api_url, headers=api_call_headers, verify=False)

print api_call_response.text